src/olympia/devhub/tests/test_tasks.py

```diff
@@ -539,6 +539,43 @@ def test_xpi_autoclose_is_enabled(self, subprocess_mock):
 
         assert '--disable-xpi-autoclose' not in self.FakePopen.get_args()
 
+    @override_switch('enable-mv3-submissions', active=False)
+    def test_mv3_submissions_waffle_disabled(self):
+        with mock.patch('olympia.devhub.tasks.subprocess') as subprocess_mock:
+            subprocess_mock.Popen = self.FakePopen
+
+            tasks.run_addons_linter(
+                path=self.valid_path, channel=amo.RELEASE_CHANNEL_LISTED
+            )
+
+            assert '--max-manifest-version=3' not in self.FakePopen.get_args()
+            assert '--max-manifest-version=2' in self.FakePopen.get_args()
+
+        mv3_path = get_addon_file('webextension_mv3.xpi')
+        result = tasks.run_addons_linter(mv3_path, channel=amo.RELEASE_CHANNEL_LISTED)
+        assert json.loads(result or '{}').get('errors') == 1
```

Comment:
if the test passes then the result will be valid.  But if the test doesn't pass then it means the error message is correct (rather than the message be a complaint that `none` can't be parsed as valid json).